app/models/derived_variables/lettings_log_variables.rb

```diff
@@ -68,6 +67,7 @@ def set_derived_fields!
         self.voiddate = startdate
       end
     end
+    self.vacdays = property_vacant_days
```

Comment:
Is the issue here that `property_vacant_days` uses `self.voiddate` so getting `vacdays` right was order of operations dependent?

Reply:
Yup

Reply:
I guess the spec helps protect against that in future, might still be less brittle to make it more functional though. I.e. if `property_vacant_days` requires `voiddate` then make that explicit and pass it in as a param?

Reply:
true, but I think we've got a few variables like that. Especially the location ones